src/cloud-functions.ts

```diff
@@ -545,3 +587,51 @@ export function optionsToTrigger(options: DeploymentOptions) {
 
   return trigger;
 }
+
+export function optionsToEndpoint(
+  options: DeploymentOptions
+): ManifestEndpoint {
+  const endpoint: ManifestEndpoint = {};
+  copyIfPresent(
+    endpoint,
+    options,
+    'minInstances',
+    'maxInstances',
+    'ingressSettings',
+    'labels',
+    'timeoutSeconds'
+  );
+  convertIfPresent(endpoint, options, 'region', 'regions', (r) => r);
+  convertIfPresent(
+    endpoint,
+    options,
+    'serviceAccountEmail',
+    'serviceAccount',
+    (sa) => sa
+  );
+  convertIfPresent(endpoint, options, 'vpc', 'vpcConnector', (connector) => {
+    return { connector };
+  });
+  convertIfPresent(
+    endpoint,
+    options,
+    'vpc',
+    'vpcConnectorEgressSettings',
+    (egressSettings, vpc) => {
```

Comment:
This is confusing to me. Is this an edit in place as an optional second parameter? I'm wondering if a more explicit approach will be clearer since now convertIfPresent only works when the calls are in the right order.

Maybe something vaguely like:

```
const vpc: <whateverthetypeishere> = {};
convertIfPresent(vpc, options, 'connector', 'vpcConnector');
convertIfPresent(vpc, options, 'egressSettings', 'vpcConnectorEgressSettings');
if (Object.keys(vpc).length !== 0) {
  endpoint.vpc = vpc;
}
```

Reply:
That's fair. I agree that the suggested code is simpler!